core/commands/cid.go

```diff
@@ -0,0 +1,307 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"unicode"
+
+	cmds "gx/ipfs/QmPTfgFTo9PFr1PvPKyKoeMgBvYPh6cX3aDP7DHKVbnCbi/go-ipfs-cmds"
+	mhash "gx/ipfs/QmPnFwZ2JXKnXgMw8CdBPxn7FWh6LLdjUjxV1fKHuJnkr8/go-multihash"
+	cidutil "gx/ipfs/QmPyxJ2QS7L5FhGkNYkNcXHGjDhvGHueJ4auqAstFHYxy5/go-cidutil"
+	cmdkit "gx/ipfs/QmSP88ryZkHSRn1fnngAaV2Vcn63WUJzAavnRM9CVdU1Ky/go-ipfs-cmdkit"
+	mbase "gx/ipfs/QmSbvata2WqNkqGtZNg8MR3SKwnB8iQ7vTPJgWqB8bC5kR/go-multibase"
+	verifcid "gx/ipfs/QmVUhfewLZpSaAiBYCpw2krYMaiVmFuhr2iurQLuRoU6sD/go-verifcid"
+	cid "gx/ipfs/QmZFbDTY9jfSBms2MchvYM9oYRbAF19K7Pby47yDBfpPrb/go-cid"
+)
+
+var CidCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Convert and discover properties of CIDs",
+	},
+	Subcommands: map[string]*cmds.Command{
+		"format": cidFmtCmd,
+		"base32": base32Cmd,
+		"bases":  basesCmd,
+		"codecs": codecsCmd,
+		"hashes": hashesCmd,
+	},
+}
+
+var cidFmtCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Format and convert a CID in various useful ways.",
+		LongDescription: `
+Format and converts <cid>'s in various useful ways.
+
+The optional format string either "prefix" or or a printf style format string:
+` + cidutil.FormatRef,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, true, "Cids to format."),
```

Comment:
Can we use `WithDefault("%s")`?

Reply:
You meant for the `-f` option?  If it is omitted that is the default (see below), I just didn't use WithDefault.  The idea was the `-f` is optional and if its not specified it will just return the CIDs.

Reply:
It's nice from a documentation standpoint.

Reply:
No Problem.